### DISCUSSION TITLE
Добавить поле Icon для выбора иконок с предпросмотром

### DISCUSSION THREAD

**Discussion by dragomano** [moonshine-software/moonshine#894] [Ideas]
Можно унаследовать от поля Select. Чтобы не искать иконки в документации, можно было бы просто сделать поиск по ключевым словам или названиям иконок в этом поле.

На примере поля [Font Awesome](https://github.com/aerni/statamic-font-awesome) в Statamic:
![sshot-8](https://github.com/moonshine-software/moonshine/assets/229402/912f7977-f089-4b09-9516-c6a5613fe457)

**Comment by DissNik** [comment]
Вы хотите создать поле в MoonShine или добавить поиск в доку?
Если поле то поле селект позволяет выводить опции с изображениями https://moonshine-laravel.com/docs/resource/fields/fields-select#with-image
передайте данные с иконками

**Reply by dragomano** [reply]
Я предлагаю идею для создания такого поля в Moonshine.

**Reply by DissNik** [reply]
иконки просто лежат в отдельных файлах, просмотрите содержимое нужной директории с иконками и сформируйте options для поля select

**Comment by dragomano** [comment]
[Реализовал](https://github.com/dragomano/moonshine-heroicons-field)

Но всё ещё теряюсь, как использовать svg, лежащие в php файлах, поэтому пока решил использовать внешний ресурс.

![sshot-25](https://github.com/moonshine-software/moonshine/assets/229402/6cc51a95-6f8a-4119-95dc-519d347ee278)

**Reply by dragomano** [reply]
Интегрировал blade-heroicons, в том пакете полный набор иконок, все 4 стиля. Теперь можно задавать стиль для каждого отдельного поля с помощью метода `style`.
